### PR TITLE
Make Leech event on successful run mandatory

### PR DESCRIPTION
**Pull request by locks**
Closes https://github.com/mtgred/netrunner/issues/5685

### CODE REVIEW

File: src/clj/game/cards/programs.clj [hunk 1]
```diff
@@ -1537,17 +1537,10 @@
...
-             :optional
-             {:player :runner
-              :req (req (is-central? (target-server context)))
-              :waiting-prompt "Runner to decide if they will use Leech"
-              :autoresolve (get-autoresolve :auto-fire)
-              :prompt "Use Leech?"
-              :yes-ability {:msg "add 1 virus counter to Leech"
-                            :effect (req (add-counter state side card :virus 1))}
-              :no-ability {:effect (req (system-msg state side "does not add counter to Leech"))}}}]
+             :interactive (get-autoresolve :auto-fire (complement never?))
```
Comment (NoahTheDuke) [inline src/clj/game/cards/programs.clj:1540]:
```suggestion
             :interactive (req true)
```

Comment (NoahTheDuke) [inline src/clj/game/cards/programs.clj:1540]: Oops, I led you astray! You can remove this and the `:silent` entirely. They're only needed with `:optional` stuff.

Comment (locks) [reply src/clj/game/cards/programs.clj]: I removed both and added `:req` which I believe is necessary? Going to take a go at tests next!

**Comment by locks** [comment]
Ready for review, I believe!

**Pull request merged**